contracts/Utils.sol

```diff
@@ -58,11 +88,17 @@ contract Utils {
     }
 
     function getDecimals(IERC20 token) internal view returns (uint256) {
-        if (token == ETH_TOKEN_ADDRESS) return ETH_DECIMALS; // save storage access
-        uint256 tokenDecimals = decimals[token];
+        // return token decimals if handled by token switch
+        uint256 tokenDecimals = tokenSwitch(token);
+        if (tokenDecimals > 0) return tokenDecimals;
+
+        // handle case where token decimals is not a declared constant
+        tokenDecimals = decimals[token];
         // moreover, very possible that old tokens have decimals 0
         // these tokens will just have higher gas fees.
-        if (tokenDecimals == 0) return token.decimals();
```

Comment:
I prefer the trenary short version

return (tokenDecimals > 0 ? tokenDecimals : token.deciamls());

what do u think?
:)

Reply:
Yeah, much preferred. Updated to this.